nvtabular/workflow.py

```diff
@@ -74,6 +76,187 @@ def __init__(self, cat_names=None, cont_names=None, label_name=None, config=None
 
         self.clear_stats()
 
+    def _register_ops(self, operators):
+        if not isinstance(operators, list):
+            operators = [operators]
+        # order matters!!!
+        if "full" not in self.columns_ctx:
+            self.columns_ctx["full"] = {}
+            # full current will always be the most up to date version of columns,
+            # based on operators added
+            self.columns_ctx["full"]["base"] = (self.columns_ctx["all"]["base"],)
+            current = (self.columns_ctx["all"]["base"],)
+        # for all what does not exist in
+        for op, col_focus, chain_on_ops, child, parent in operators:
+            target_cols = op.get_columns(self.columns_ctx, col_focus, chain_on_ops)
+            # grab the input target columns and the input extra columns
+            extra_cols = []
+            cur_extra_cols = [] if len(current) < 2 else current[1]
+            full_list = current[0] + cur_extra_cols
+            for col in full_list:
+                if col not in target_cols:
+                    extra_cols.append(col)
+            current = self.create_full_col_ctx_entry(op, target_cols, extra_cols, parent=parent)
+        self.reduce(self.columns_ctx["full"])
+
+    def reduce(self, full_dict):
```

Comment:
can  we make it clear that these new methods on the workflow object are private - so users don't need to care about them. L

Methods like reduce/transform/analyze_placement/detect_cols_collision etc should also have a '_' by them ?